homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -115,21 +136,36 @@ def icon(self):
         """Icon to use in the frontend, if any."""
         return ICON
 
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._type == "inverters":
+            return {"last_reported": self._last_reported}
+
+        return None
+
     async def async_update(self):
         """Get the energy production data from the Enphase Envoy."""
-
         if self._type != "inverters":
-            _state = await getattr(EnvoyReader(self._ip_address), self._type)()
+            _state = await getattr(self._envoy_reader, self._type)()
             if isinstance(_state, int):
                 self._state = _state
             else:
                 _LOGGER.error(_state)
                 self._state = None
 
-        elif self._type == "inverters":
-            inverters = await (EnvoyReader(self._ip_address).inverters_production())
+        elif self._type == "inverters" and self._inv_auth:
+            try:
+                inverters = await (self._envoy_reader.inverters_production())
+            except requests.exceptions.HTTPError:
+                _LOGGER.warning(
+                    "Authentication for Inverter data failed during update: %s",
```

Comment:
We already checked authentication in `setup_platform`. When can it fail here? Maybe it can fail for another reason here?

Reply:
I was thinking if the username or password was changed on the device itself, then a failure could occur during `async_update()`.  Kind of a not so common case, should I remove this try/except block?

Reply:
Ok, let's keep it then.